eidas-saml-parser/src/main/java/uk/gov/ida/notification/eidassaml/EidasSamlResource.java

```diff
@@ -75,4 +96,34 @@ private String getLevelsOfAssurance(AuthnRequest authnRequest) {
                 .map(AuthnContextClassRef::getAuthnContextClassRef)
                 .collect(Collectors.joining(","));
     }
+
+    private Credential getSigningCredential(String base64X509Certificate) throws CertificateException {
+        CertificateFactory certificateFactory = CertificateFactory.getInstance("X509");
+
+        byte[] decodedCert = Base64.getMimeDecoder().decode(base64X509Certificate);
+        ByteArrayInputStream inputStream = new ByteArrayInputStream(decodedCert);
+        Certificate certificate = certificateFactory.generateCertificate(inputStream);
+
+        BasicCredential signingCredential = new BasicCredential(certificate.getPublicKey());
+        signingCredential.setUsageType(UsageType.SIGNING);
+
+        return signingCredential;
+    }
+
+    private void validateIssuer(AuthnRequest authnRequest) {
+        if (authnRequest.getIssuer().getValue() == null) {
+            throw new SamlTransformationErrorException("Issuer is missing", Level.WARN);
+        }
+    }
+
+    private void isRequestSignatureValid(AuthnRequest authnRequest, Credential signingCredential) {
```

Comment:
Is it worth moving the logic about signatures and credential retrieval etc to some sort of helper or validator? A bit like the AssertionConsumerServiceValidator below maybe?

Reply:
Credentials I think we should hold off on until we've considered any changes we might make to the `CountryMetadataResponse` (i.e. whether we decode the base64 when it comes in and present an actual certificate in the DTO - maybe we also present the credentials or present those instead?).

Signature validation; as discussed in the office, there's a better candidate for that.